progress/README.md

```diff
@@ -0,0 +1,256 @@
+# Progress
+
+## **Requests**
+
+|                  Route                   |                         Description                          |
+| :--------------------------------------: | :----------------------------------------------------------: |
+|        [GET /progress](#get-post)        | Retrieves progress entries based on the provided parameters. |
+|     [POST /progress](#post-progress)     |                Creates a new progress entry.                 |
+| [DELETE /progress/:id](#delete-progress) |       Deletes a progress entry with the specified ID.        |
+
+## **GET /progress**
+
+Retrieves progress entries based on the provided parameters.
+
+- **Params**
+  None
+- **Query**  
+  type (query parameter): Specifies the type of progress to filter (e.g., "task", "user").
+  id (query parameter): Specifies the ID of the progress entry to retrieve (optional).
+- **Body**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  None
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** 
+    ```json
+		{
+      "message": "User progress retrieved successfully.",
+      "progress": [
+        { "<user_progress_object_1>" },
+        { "<user_progress_object_2>" },
+        ..
+      ]
+    }
+    ```
+- **Error Response:**
+  - **Code:** 500
+    - **Content:** `{ 'statusCode': 500, 'error': 'Internal Server Error', 'message': 'An internal server error occurred' }`
+
+- **Example:**
+  GET /progress?type=task
+    ```json
+    {
+			"message": "All User Progress found successfully.",
+			"progress": [
+				{
+				"id": "progress1",
+				"type": "task",
+				"userId": "b0ad77cf461633s7745d59d8",
+				"taskId": "d576s3cd4450df7698731b7a",
+				"completed": "added modal progress",
+				"planned": "write test for modal progress",
+				"blockers": "None",
+				"createdAt": 1654101900000
+				},
+				{
+				"id": "progress2",
+				"type": "user",
```

Comment:
I think for  `GET /progress?type=task`, the response will only contain objects with `type: "task"`.
Maybe I am missing something

Reply:
+1